### PR TITLE
[Kodi] Added new "ExecuteAction" Channel

### PR DESCRIPTION
**Pull request by AnthraxLeprosyPi**
Added a new advanced Channel called ExecuteAction which allows to send on of the Input.Actions described [here](http://kodi.wiki/view/JSON-RPC_API/v8#Input.Action).

This allows for even more possibilities to control Kodi.

The motivation for this was to control volume via CEC (see example)...
Example: HOMETHEATER_ExecuteAction.sendCommand("volumeup")

Signed-off-by: AnthraxLeprosyPi <anthrax.leprosy.pi@gmail.com>

### CODE REVIEW

File: addons/binding/org.openhab.binding.kodi/ESH-INF/i18n/kodi_de.properties [hunk 1]
```diff
@@ -53,6 +53,8 @@ channel-type.kodi.input.label = Tastendruck
 channel-type.kodi.input.description = Erm�glicht das Senden eines Eingabebefehls an das Kodi Media Center.
 channel-type.kodi.inputtext.label = Texteingabe
 channel-type.kodi.inputtext.description = Erm�glicht das Senden eines Eingabetextes an das Kodi Media Center.
+channel-type.kodi.inputexecuteaction.label = Aktion
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.kodi/ESH-INF/i18n/kodi_de.properties]: Should be `channel-type.kodi.inputaction.label`.

File: addons/binding/org.openhab.binding.kodi/ESH-INF/i18n/kodi_de.properties [hunk 2]
```diff
@@ -53,6 +53,8 @@ channel-type.kodi.input.label = Tastendruck
...
+channel-type.kodi.inputexecuteaction.description = Erm�glicht das Senden einer auszuf�hrenden vordefinierten Aktion an das Kodi Media Center.
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.kodi/ESH-INF/i18n/kodi_de.properties]: Should be `channel-type.kodi.inputaction.description`.

File: addons/binding/org.openhab.binding.kodi/ESH-INF/thing/thing-types.xml [hunk 3]
```diff
@@ -20,6 +20,7 @@
 			<channel id="shownotification" typeId="shownotification" />
 			<channel id="input" typeId="input" />
 			<channel id="inputtext" typeId="inputtext" />
+			<channel id="inputaction" typeId="inputexecuteaction" />
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.kodi/ESH-INF/thing/thing-types.xml]: `typeId="inputaction"`

File: addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/KodiBindingConstants.java [hunk 4]
```diff
@@ -50,6 +50,8 @@
 
     public static final String CHANNEL_INPUT = "input";
     public static final String CHANNEL_INPUTTEXT = "inputtext";
+    public static final String CHANNEL_INPUTEXECUTEACTION = "inputexecuteaction";
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/KodiBindingConstants.java]: `public static final String CHANNEL_INPUTACTION = "inputaction";`

File: addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/handler/KodiHandler.java [hunk 5]
```diff
@@ -198,6 +198,14 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                     updateState(CHANNEL_INPUTTEXT, UnDefType.UNDEF);
                 }
                 break;
+            case CHANNEL_INPUTEXECUTEACTION:
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/handler/KodiHandler.java]: see above

**Comment by AnthraxLeprosyPi** [comment]
Man, you guys have eagle eyes ;-)

File: addons/binding/org.openhab.binding.kodi/README.md [hunk 6]
```diff
@@ -75,12 +75,28 @@ The Kodi thing supports the following channels:
...
-| input            | String    | Allows to control Kodi. Valid values are: `Up`, `Down`, `Left`, `Right`, `Select`, `Back`, `Home`, `ContextMenu`, `Info`, `ShowOSD`, `ShowPlayerProcessInfo`, `SendText` and `ExecuteAction` |
+| input            | String    | Allows to control Kodi. Valid values are: `Up`, `Down`, `Left`, `Right`, `Select`, `Back`, `Home`, `ContextMenu`, `Info`, `ShowOSD` and `ShowPlayerProcessInfo`                              |
 | inputtext        | String    | This channel emulates a keyboard input                                                                                                                                                       |
+| inputaction      | String    | Send a predefined action to the Kodi Player to control the UI and/or perform other tasks. Valid values are: `left`, `right`, `up`, `down`, `pageup`, `pagedown`, `select`, `highlight`,      |
+|                                `parentdir`, `parentfolder`, `back`, `menu`, `previousmenu`, `info`, `pause`, `stop`, `skipnext`, `skipprevious`, `fullscreen`, `aspectratio`, `stepforward`, `stepback`, `bigstepforward`,  |
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.kodi/README.md]: This won't work. If you click an the `View` button next to this file in GitHub you will see a preview. The table doesn't look nice.

To make @martinvw happy you should format the tables in the `README.md` file.

http://www.tablesgenerator.com/markdown_tables

Select 'File' > 'Past table data ...' and past the unformatted mark down table, you can then copy paste the formatted table. :+1:

**Comment by AnthraxLeprosyPi** [comment]
To be honest, this is getting tedious...

Takes more time/effort to get the PR through than to develop the damn thing.

**Comment by AnthraxLeprosyPi** [comment]
Allright guys, 

I give up. last commit now fails because I edited via github webpage and seem to have forgotten to sign-off... 

I can't deal with this anymore... 

have fun and feel free to use whatever code I commited in the past.

best regards,
Anthrax

**Comment by martinvw** [comment]
Thanks!

**Pull request merged**